ui/src/components/chat/pinned_messages/mod.rs

```diff
@@ -0,0 +1,252 @@
+use common::{
+    language::get_local_text,
+    state::{Chat, Identity, State},
+    warp_runner::{
+        thumbnail_to_base64,
+        ui_adapter::{self, Message},
+        RayGunCmd, WarpCmd,
+    },
+    WARP_CMD_CH,
+};
+use dioxus::prelude::*;
+
+use dioxus_desktop::use_eval;
+use futures::StreamExt;
+use kit::components::{embeds::file_embed::FileEmbed, message::ChatText, user_image::UserImage};
+use uuid::Uuid;
+use warp::{logging::tracing::log, raygun::PinState};
+
+enum ChannelCommand {
+    FetchPinnedMessages(Uuid, usize, usize),
+    RemovePinnedMessage(Uuid, Uuid),
+}
+
+#[derive(Props, Eq, PartialEq)]
+pub struct Props {
+    #[props(!optional)]
+    active_chat: Option<Chat>,
+}
+
+#[allow(non_snake_case)]
+pub fn PinnedMessages(cx: Scope<Props>) -> Element {
+    log::trace!("rendering pinned_messages");
+    let state = use_shared_state::<State>(cx)?;
+    let _loading = use_state(cx, || true);
+    let newely_fetched_messages: &UseRef<Option<(Uuid, Vec<ui_adapter::Message>, bool)>> =
```

Comment:
I'm not sure that `PinnedMessages` needs to worry about `newely_fetched_messages`. I believe this and the related code was erroneously copied from `compose/messages.rs`

Reply:
see comment https://github.com/Satellite-im/Uplink/pull/1121#discussion_r1294466623

Reply:
remove that whole part as pinned msg are now fetched at once